silverback/persistence.py

```diff
@@ -0,0 +1,262 @@
+import pickle
+from abc import ABC, abstractmethod
+from datetime import datetime, timezone
+from typing import TYPE_CHECKING, Annotated, Any, Dict, Optional, Type
+
+from ape.logging import logger
+from pydantic import BaseModel
+from taskiq import TaskiqResult
+from typing_extensions import Self  # Introduced 3.11
+
+from .types import SilverbackIdent
+
+
+class SilverbackState(BaseModel):
+    instance: str
+    network: str
+    # Last block number seen by runner
+    last_block_seen: int
+    # Last block number processed by a worker
+    last_block_processed: int
+    updated: datetime
+
+
+class HandlerResult(BaseModel):
+    instance: str
+    network: str
+    handler_id: str
+    block_number: Optional[int]
+    log_index: Optional[int]
+    execution_time: float
+    # TODO: upcoming feature in taskiq
+    # labels: Dict[str]
+    # TODO: Use computed field with pydantic v2
+    return_value_blob: Optional[bytes]  # pickled data
+    created: datetime
+
+    @classmethod
+    def from_taskiq(
+        cls,
+        ident: SilverbackIdent,
+        handler_id: str,
+        block_number: int | None,
+        log_index: int | None,
+        result: TaskiqResult,
+    ) -> Self:
+        return cls(
+            instance=ident.identifier,
+            network=ident.network_choice,
+            handler_id=handler_id,
+            block_number=block_number,
+            log_index=log_index,
+            execution_time=result.execution_time,
+            # labels=result.labels,
+            return_value_blob=pickle.dumps(result.return_value),
+            created=datetime.now(timezone.utc),
+        )
+
+
+class BasePersistentStorage(ABC):
```

Comment:
Is there anything we need to add for v1?  This is bare minimum for what this PR needs but we could future-think here as well for aggregate data or API fetching.